core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -487,6 +490,18 @@ private[spark] object SparkHadoopUtil extends Logging {
     if (conf.getOption("spark.hadoop.fs.s3a.downgrade.syncable.exceptions").isEmpty) {
       hadoopConf.set("fs.s3a.downgrade.syncable.exceptions", "true")
     }
+    // In Hadoop 3.3.1, AWS region handling with the default "" endpoint only works
+    // in EC2 deployments or when the AWS CLI is installed.
+    // The workaround is to set the name of the S3 endpoint explicitly,
```

Comment:
We can use `AWS_REGION` as a workaround, too.

Reply:
that's automatically picked up in the default chain, but since you have to set it everywhere it's not something you can rely on.